keras_cv/layers/preprocessing/cut_mix.py

```diff
@@ -0,0 +1,149 @@
+
+    def call(self, images, labels):
+        """
+        call method for the CutMix layer.
+
+        Args:
+            images: Tensor representing images of shape [batch_size, width, height, channels].
+            labels: One hot encoded tensor of labels for the images.
+        Returns:
+            images: augmented images, same shape as input.
+            labels: updated labels with both label smoothing and the cutmix updates applied.
+        """
+        augment_cond = tf.less(
+            tf.random.uniform(shape=[], minval=0.0, maxval=1.0), self.probability
+        )
+        # pylint: disable=g-long-lambda
+        cutmix_augment = lambda: self._update_labels(*self._cutmix(images, labels))
+        no_augment = lambda: (images, self._smooth_labels(labels))
+        return tf.cond(augment_cond, cutmix_augment, no_augment)
+
+    def _cutmix(self, images, labels):
+        """Apply cutmix."""
+        input_shape = tf.shape(images)
+        batch_size, image_height, image_width = (
+            input_shape[0],
+            input_shape[1],
+            input_shape[2],
+        )
+
+        lambda_sample = CutMix._sample_from_beta(
+            self.alpha, self.alpha, (tf.shape(labels)[0],)
+        )
+
+        ratio = tf.math.sqrt(1 - lambda_sample)
+
+        cut_height = tf.cast(
+            ratio * tf.cast(image_height, dtype=tf.float32), dtype=tf.int32
+        )
+        cut_width = tf.cast(
+            ratio * tf.cast(image_height, dtype=tf.float32), dtype=tf.int32
+        )
+
+        random_center_height = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_height, dtype=tf.int32
+        )
+        random_center_width = tf.random.uniform(
+            shape=[batch_size], minval=0, maxval=image_width, dtype=tf.int32
+        )
+
+        bbox_area = cut_height * cut_width
+        lambda_sample = 1.0 - bbox_area / (image_height * image_width)
+        lambda_sample = tf.cast(lambda_sample, dtype=tf.float32)
+
+        images = tf.map_fn(
+            lambda x: _fill_rectangle(*x),
+            (
+                images,
+                random_center_width,
+                random_center_height,
+                cut_width // 2,
+                cut_height // 2,
+                tf.reverse(images, [0]),
+            ),
+            dtype=(tf.float32, tf.int32, tf.int32, tf.int32, tf.int32, tf.float32),
+            fn_output_signature=tf.TensorSpec(images.shape[1:], dtype=tf.float32),
+        )
+
+        return images, labels, lambda_sample
+
+    def _update_labels(self, images, labels, lambda_sample):
+        labels_1 = self._smooth_labels(labels)
+        cutout_labels = tf.reverse(labels_1, [0])
+
+        lambda_sample = tf.reshape(lambda_sample, [-1, 1])
+        labels = lambda_sample * labels_1 + (1.0 - lambda_sample) * cutout_labels
+
+        return images, labels
+
+    def _smooth_labels(self, labels):
+        label_smoothing = self.label_smoothing or 0.0
```

Comment:
Any reason for the or check here? Are you trying to filter None/False?

Reply:
Originally I was going to have label_smoothing default to `None`, we can probably just remove the `or` clause here.   WDYT?